libr/core/cmd_search.c

```diff
@@ -1549,6 +1549,143 @@ static void do_esil_search(RCore *core, struct search_parameters *param, const c
 	r_cons_clear_line (1);
 }
 
+static void do_syscall_search(RCore *core, struct search_parameters *param) {
+	RSearch *search = core->search;
+	ut64 at;
+	ut64 curpc ;
+	ut8 *buf;
+	ut8 *arr;
+	RAnalOp aop;
+	int i,j,inslen, ret, bsize = R_MIN (64, core->blocksize);
+	int kwidx = core->search->n_kws;
+	int count = 0;	
+	char *input = "swi";
+	char temp_inst[50] ;
+	int nb_opcodes = 8 ;
+	ut64 old_offset = core->offset;
+	int nbytes = 0;
+	RAnalEsil *ESIL = core->anal->esil;
+
+	r_core_cmd0 (core, "aei");
```

Comment:
that will break the debugger and the current esil state, by using the api you can just create your own esil state and save/restore the regstate. ae apis are messing with many side effects that are probably undesired for a search operation. (memory read, write..)